### PR TITLE
[ws-daemon] don't fail if prebuilt is not found

### LINKED ISSUE #2525: My workspace is not starting
### Describe the bug
Error msg rcvd: "Last run 3 minutes ago: cannot initialize workspace: remote content error"
Workspaces-Gitpod-error

### PR DESCRIPTION
**Pull request by AlexTugarev**
should fix #2525

### CODE REVIEW

File: components/ws-daemon/pkg/content/initializer.go [hunk 1]
```diff
@@ -60,7 +67,7 @@ func collectRemoteContent(ctx context.Context, rs storage.DirectAccess, ps stora
 		}
 		info, err := ps.SignDownload(ctx, bkt, obj)
 		if err != nil {
-			return nil, xerrors.Errorf("cannot find snapshot: %w", err)
+			return nil, ErrCannotFindSnapshot
```
Comment (AlexTugarev) [inline components/ws-daemon/pkg/content/initializer.go:70]: @csweichel that's what you meant? 🤷🏻‍♂️

Comment (csweichel) [reply components/ws-daemon/pkg/content/initializer.go:70]: Looks good. Only thing: we should check if the error actually is `storage.ErrNotFound`

File: components/ws-daemon/pkg/content/initializer.go [hunk 2]
```diff
@@ -41,6 +42,12 @@ type RunInitializerOpts struct {
 	OWI map[string]interface{}
 }
 
+// errors to be tested with errors.Is
+var (
+	// cannot find snapshot
+	ErrCannotFindSnapshot = errors.New("cannot find snapshot")
```
Comment (csweichel) [inline components/ws-daemon/pkg/content/initializer.go:48]: `ErrCannotFindSnapshot` does not necessarily be public (i.e. start with an uppercase letter), it's only used within the same package.

Comment (AlexTugarev) [reply components/ws-daemon/pkg/content/initializer.go:48]: ah!

File: components/ws-daemon/pkg/content/initializer.go (hunk 1 above)
Comment (AlexTugarev) [reply components/ws-daemon/pkg/content/initializer.go:70]: ok

Comment (AlexTugarev) [reply components/ws-daemon/pkg/content/initializer.go]: can there be any other error otherwise? so, should the other error check remain?

Comment (csweichel) [reply components/ws-daemon/pkg/content/initializer.go]: Yes and Yes :)

File: components/ws-daemon/pkg/content/service.go [hunk 3]
```diff
@@ -195,6 +196,10 @@ func (s *WorkspaceService) InitWorkspace(ctx context.Context, req *api.InitWorks
 		}
 
 		remoteContent, err := collectRemoteContent(ctx, rs, ps, workspace.Owner, req.Initializer)
+		if err != nil && errors.Is(err, errCannotFindSnapshot) {
+			log.WithError(err).Error("cannot find snapshot")
+			return nil, status.Error(codes.Internal, "cannot find snapshot")
```
Comment (csweichel) [inline components/ws-daemon/pkg/content/service.go:201]: we could return `codes.NotFound` here

**Review by csweichel — APPROVED** [review]
LGTM

**Pull request merged**